File: lib/ml/hyperheap/FStar_ST.ml
type 'a __ref = 'a ref
type 'a ref = 'a __ref
let read x = !x
let op_Colon_Equals = (fun i r v -> r := v)
let op_Bang = (fun i r -> !r)
let alloc = (fun x -> ref x)
let new_region = (fun r0 -> ())
let ralloc = (fun i x -> ref x)
let recall = (fun i r -> ())

File: stage0/ulib/fs/FStar_ST.fs
#light "off"
module FStar_ST

open FStar_CommonST

type 'a mref = 'a FStar_Monotonic_Heap.mref
type 'a ref = 'a FStar_Monotonic_Heap.ref

let read = read
let op_Bang = op_Bang

let write = write
let op_Colon_Equals = op_Colon_Equals

let alloc x = alloc

let recall = recall
let get = get

type 'a witnessed = 'a FStar_CommonST.witnessed

let gst_Witness = gst_witness
let gst_recall = gst_recall